flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBStateBackend.java

```diff
@@ -276,6 +312,12 @@ private RocksDBStateBackend(RocksDBStateBackend original, Configuration config)
 		this.enableIncrementalCheckpointing = original.enableIncrementalCheckpointing.resolveUndefined(
 			config.getBoolean(CheckpointingOptions.INCREMENTAL_CHECKPOINTS));
 
+		if (original.restoredThreadNum == CheckpointingOptions.CHECKPOINT_RESTORE_THREAD_NUM.defaultValue()) {
```

Comment:
here we could initialise `restoredThreadNum` from `config` if `original.restoredThreadNum` is -1 (undefined). Otherwise `restoredThreadNum` = `original.restoredThreadNum`, similar to `original.enableIncrementalCheckpointing.resolveUndefined`.